test_neighbor_mac_noptf failing in multi asic devices

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/Azure/SONiC/wiki#report-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- SONiC Support Forums - https://groups.google.com/forum/#!forum/sonicproject

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**

<!--
Briefly describe the problem you are having in a few paragraphs.
-->
The test in file `arp/test_neighbor_mac_noptf` are failing on multi asic platforms. The tests fail with following error
```
self = <tests.arp.test_neighbor_mac_noptf.TestNeighborMacNoPtf instance at 0x7f11261d6780>
duthosts = <tests.common.devices.duthosts.DutHosts object at 0x7f1125516250>
enum_rand_one_per_hwsku_frontend_hostname = 'svcstr-nsonic-acs-1'

    @pytest.fixture(scope="module", autouse=True)
    def setupDutConfig(self, duthosts, enum_rand_one_per_hwsku_frontend_hostname):
        """
            Disabled BGP to reduce load on switch and restores DUT configuration after test completes
    
            Args:
                duthost (AnsibleHost): Device Under Test (DUT)
    
            Returns:
                None
        """
        duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
        duthost.command("sudo config bgp shutdown all")
        if not wait_until(120, 2.0, self._check_no_bgp_routes, duthost):
>           pytest.fail('BGP Shutdown Timeout: BGP route removal exceeded 120 seconds.')
E           Failed: BGP Shutdown Timeout: BGP route removal exceeded 120 seconds.

duthost    = <MultiAsicSonicHost> svcstr-nsonic-acs-1
duthosts   = <tests.common.devices.duthosts.DutHosts object at 0x7f1125516250>
enum_rand_one_per_hwsku_frontend_hostname = 'svcstr-nsonic-acs-1'
self       = <tests.arp.test_neighbor_mac_noptf.TestNeighborMacNoPtf instance at 0x7f11261d6780>

```
The PR https://github.com/Azure/sonic-mgmt/pull/3369 introduced the change to count the routes. This change doesn't work on multi asic platform.

--- a/tests/arp/test_neighbor_mac_noptf.py
+++ b/tests/arp/test_neighbor_mac_noptf.py
@@ -30,22 +30,33 @@ class TestNeighborMacNoPtf:
         6: {"intfIp": "fe00::1/64", "NeighborIp": "fe00::2"},
     }	
 
-    def count_routes(self, host, prefix):
+    def count_routes(self, asichost, prefix):
         # Counts routes in ASIC_DB with a given prefix
-        num = host.shell(
-                'sonic-db-cli ASIC_DB eval "return #redis.call(\'keys\', \'{}:{{\\"dest\\":\\"{}*\')" 0'.format(ROUTE_TABLE_NAME, prefix),
+        num = asichost.shell(
+                '{} ASIC_DB eval "return #redis.call(\'keys\', \'{}:{{\\"dest\\":\\"{}*\')" 0'.format(asichost.sonic_db_cli,ROUTE_TABLE_NAME, prefix),
                 module_ignore_errors=True, verbose=True)['stdout']
         return int(num)
 
+    def _check_no_bgp_routes_asic(self, asichost):
+        # Checks that there are no routes installed by BGP in ASIC_DB by filtering out all local routes installed on asic
+        localv6 = self.count_routes(asichost, "fc") + self.count_routes(asichost, "fe")
+        localv4 = self.count_routes(asichost, "10.") + self.count_routes(asichost, "192.168.0.")
+        #these routes are present only on multi asic device, on single asic platform they will be zero
+        internal = self.count_routes(asichost, "8.") + self.count_routes(asichost, "2603")
+        allroutes = self.count_routes(asichost, "")
+        logger.info("asic[{}] localv4 routes {} localv6 routes {} internalv4 {} allroutes {}".format(asichost.asic_index, localv4, localv6, internal, allroutes))
+        bgp_routes_asic = allroutes - localv6 - localv4 - internal - DEFAULT_ROUTE_NUM
+
+        return bgp_routes_asic
+
     def _check_no_bgp_routes(self, duthost):
-        # Checks that there are no routes installed by BGP in ASIC_DB by filtering out all local routes installed on testbeds
-        localv6 = self.count_routes(duthost, "fc") + self.count_routes(duthost, "fe")
-        localv4 = self.count_routes(duthost, "10.") + self.count_routes(duthost, "192.168.0.")
-        allroutes = self.count_routes(duthost, "")
-        bgproutes = allroutes - localv6 - localv4 - DEFAULT_ROUTE_NUM
-
-        return bgproutes == 0
-
+        bgp_routes = 0
+        # Checks that there are no routes installed by BGP in ASIC_DB by filtering out all local routes installed on testbed
+        for asic in duthost.asics:
+            bgp_routes += self._check_no_bgp_routes_asic(asic)
+        
+        return bgp_routes == 0
+            
     @pytest.fixture(scope="module", autouse=True)
     def setupDutConfig(self, duthosts, enum_rand_one_per_hwsku_frontend_hostname):
         """

--- a/tests/arp/test_neighbor_mac_noptf.py
+++ b/tests/arp/test_neighbor_mac_noptf.py
@@ -33,15 +33,15 @@ class TestNeighborMacNoPtf:
     def count_routes(self, asichost, prefix):
         # Counts routes in ASIC_DB with a given prefix
         num = asichost.shell(
-                '{} ASIC_DB eval "return #redis.call(\'keys\', \'{}:{{\\"dest\\":\\"{}*\')" 0'.format(asichost.sonic_db_cli,ROUTE_TABLE_NAME, prefix),
+                '{} ASIC_DB eval "return #redis.call(\'keys\', \'{}:{{\\"dest\\":\\"{}*\')" 0'.format(asichost.sonic_db_cli, ROUTE_TABLE_NAME, prefix),
                 module_ignore_errors=True, verbose=True)['stdout']
         return int(num)
 
-    def _check_no_bgp_routes_asic(self, asichost):
-        # Checks that there are no routes installed by BGP in ASIC_DB by filtering out all local routes installed on asic
+    def _get_bgp_routes_asic(self, asichost):
+        # Get the routes installed by BGP in ASIC_DB by filtering out all local routes installed on asic
         localv6 = self.count_routes(asichost, "fc") + self.count_routes(asichost, "fe")
         localv4 = self.count_routes(asichost, "10.") + self.count_routes(asichost, "192.168.0.")
-        #these routes are present only on multi asic device, on single asic platform they will be zero
+        # these routes are present only on multi asic device, on single asic platform they will be zero
         internal = self.count_routes(asichost, "8.") + self.count_routes(asichost, "2603")
         allroutes = self.count_routes(asichost, "")
         logger.info("asic[{}] localv4 routes {} localv6 routes {} internalv4 {} allroutes {}".format(asichost.asic_index, localv4, localv6, internal, allroutes))
@@ -53,7 +53,7 @@ class TestNeighborMacNoPtf:
         bgp_routes = 0
         # Checks that there are no routes installed by BGP in ASIC_DB by filtering out all local routes installed on testbed
         for asic in duthost.asics:
-            bgp_routes += self._check_no_bgp_routes_asic(asic)
+            bgp_routes += self._get_bgp_routes_asic(asic)
         
         return bgp_routes == 0
             

